entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/repository/EntityRecordRepository.java

```diff
@@ -91,7 +101,7 @@ public long deleteForGood(String entityId) {
      */
     public EntityRecord save(EntityRecord entityRecord){
         //if the id of the entity is null, get it from the database before saving  
-    	long dbId = generateAutoIncrement(entityRecord.getEntity().getType());
+    	long dbId = generateAutoIncrement(entityRecord.getClass().getInterfaces()[0].getSimpleName());
```

Comment:
please revert back this change

Reply:
Merged new changes from the develop branch with the ObjectId type used as a db index.